lib/saluki-components/src/sources/checks/python_scheduler.rs

```diff
@@ -0,0 +1,486 @@
+                    if let Some(traceback) = e.traceback_bound(py) {
+                        error!("Traceback: {}", traceback.format().expect("Could format traceback"));
+                    }
+                    return Err(generic_error!("Could not import 'datadog_checks' module"));
+                }
+            };
+            match modd.getattr("AgentCheck") {
+                Ok(c) => {
+                    agent_check_base_class = Some(c.unbind());
+                }
+                Err(e) => {
+                    error!(%e, "Could not get AgentCheck class.");
+                    if let Some(traceback) = e.traceback_bound(py) {
+                        error!("Traceback: {}", traceback.format().expect("Could format traceback"));
+                    }
+                    return Err(generic_error!("Could not find 'AgentCheck' class"));
+                }
+            };
+            info!("Found all pre-requisites for Agent python check execution");
+
+            Ok(())
+        })?;
+
+        Ok(Self {
+            tlm,
+            running: HashMap::new(),
+            agent_check_base_class: agent_check_base_class.expect("AgentCheck class should be present"),
+        })
+    }
+
+    // See `CheckRequest::to_runnable_request` for more TODO items here
+    // Returns an opaque handle to the python class implementing the check
+    fn register_check(&mut self, check: &RunnableCheckRequest) -> Result<CheckHandle, GenericError> {
+        let check_handle = match self.register_check_impl(check) {
+            Ok(h) => h,
+            Err(e) => {
+                error!(%e, "Could not register check {}", check.check_request.name);
+                return Err(e);
+            }
+        };
+        // TODO What 'init' is needed after grabbing a handle to the check class?
+        // See `rtloader/three/three.cpp::getCheck` for calls:
+        // - `AgentCheck.load_config(init_config)`
+        // - `AgentCheck.load_config(instance)`
+
+        // JK load_config is just yaml parsing -- str -> pyAny
+        // I assume my impl of CheckRequest::to_pydict makes this un-neccessary
+        // but TBD, could be some subtle differences between it and 'load_config'
+
+        // - set attr 'check_id' equal to the check id
+        // also parse out the attr `__version__` and record it somewhere
+        // ref PythonCheckLoader::Load in Agent
+
+        Ok(check_handle)
+    }
+
+    fn possible_import_paths_for_module(&self, module_name: &str) -> Vec<String> {
+        vec![module_name.to_string(), format!("datadog_checks.{}", module_name)]
+    }
+
+    fn register_check_impl(&mut self, check: &RunnableCheckRequest) -> Result<CheckHandle, GenericError> {
+        let check_module_name = &check.check_request.name;
+        // if there is a specific source, then this will populate into locals and can be found
+        if let Some(py_source) = &check.check_source_code {
```

Comment:
as noted elsewhere, this only exists for tests at this point, "production" usage of checks do not ship python source code, its always a module loaded